forc/src/cli/commands/test.rs

```diff
@@ -49,21 +52,27 @@ pub struct TestPrintOpts {
 }
 
 pub(crate) fn exec(cmd: Command) -> ForcResult<()> {
-    if let Some(ref _filter) = cmd.filter {
-        forc_result_bail!("unit test filter not yet supported");
-    }
-
     let test_runner_count = match cmd.test_threads {
         Some(runner_count) => TestRunnerCount::Manual(runner_count),
         None => TestRunnerCount::Auto,
     };
 
     let test_print_opts = cmd.test_print.clone();
+    let test_filter_phrase = cmd.filter.clone();
+    let test_filter = test_filter_phrase.as_ref().map(|filter_phrase| TestFilter {
+        filter_phrase,
+        exact_match: cmd.exact,
+    });
     let opts = opts_from_cmd(cmd);
     let built_tests = forc_test::build(opts)?;
     let start = std::time::Instant::now();
-    info!("   Running {} tests", built_tests.test_count());
-    let tested = built_tests.run(test_runner_count)?;
+    let test_count = built_tests.test_count(test_filter.as_ref());
+    info!(
+        "   Running {} tests, filtered {} tests",
+        test_count.total - test_count.filtered,
+        test_count.filtered
+    );
```

Comment:
nit: for nicer UX, print `1 test` instead of `1 tests`

```
fn formatted_test_count_string(count: usize) -> String {
    format!("{} {}", count, if count == 1 { "test" } else { "tests" })
}
```

```suggestion
    info!(
        "   Running {}, ignored {}",
        formatted_test_count_string(test_count.total - test_count.filtered),
        formatted_test_count_string(test_count.filtered),
    );
```